remote-method/index.js

```diff
@@ -0,0 +1,249 @@
+'use strict';
+var yeoman = require('yeoman-generator');
+var chalk = require('chalk');
+
+var actions = require('../lib/actions');
+var helpers = require('../lib/helpers');
+var validateName = helpers.validateName;
+var typeChoices =helpers.getTypeChoices;
+var accepts = [];
+var returns = [];
+
+module.exports = yeoman.generators.Base.extend({
+  // This generator does not track file changes via yeoman,
+  // as loopback-workspace is editing (modifying) files when
+  // saving project changes.
+  constructor: function() {
+    yeoman.generators.Base.apply(this, arguments);
+
+    this.argument('methodName', {
+      desc: 'Name of the remote method',
+      required: false,
+      type: String
+    });
+  },
+
+  help: function() {
+    return helpers.customHelp(this);
+  },
+
+  loadProject: actions.loadProject,
+
+  loadModels: actions.loadModels,
+
+  askForModel: function() {
+    // if (this.options.modelName) {
+    //   this.modelName = this.options.modelName;
+    //   return;
+    // }
+
+    var done = this.async();
+    var prompts = [
+      {
+        name: 'model',
+        message: 'Select the model:',
+        type: 'list',
+        choices: this.modelNames
+      }
+    ];
+
+    this.prompt(prompts, function(answers) {
+      this.modelName = answers.model;
+      done();
+    }.bind(this));
+  },
+
+  findModelDefinition: function() {
+    this.modelDefinition = this.projectModels.filter(function(m) {
+      return m.name === this.modelName;
+    }.bind(this))[0];
+
+    if (!this.modelDefinition) {
+      var msg = 'Model not found: ' + this.modelName;
+      this.log(chalk.red(msg));
+      this.async()(new Error(msg));
+    }
+  },
+
+  askForParameters: function() {
+    var done = this.async();
+    var name = this.methodName;
+    var prompts = [
+      {
+        name: 'methodName',
+        message: 'Enter the remote method name:',
+        required: true,
+        default: name,
+        validate: function(value) {
+          return validateName(value);
+        }
+      },
+      {
+        name: 'isStatic',
+        message: 'Is Static?',
+        required:true,
+        type: 'confirm',
+        default: true
+      },
+      {
+        name: 'description',
+        message: 'Description for method:'
+      },
+      {
+        name: 'httpPath',
+        message: 'HTTP path:',
+        required: true
+      },
+      {
+        name: 'httpVerb',
+        message: 'HTTP verb:',
+        type: 'list',
+        choices: ['post', 'put', 'delete', 'get', 'head', 'patch', 'other']
```

Comment:
We use `null` for custom value in other places (see e.g. `getTypeChoices`), please stay consistent and use the same approach here too.

```js
choices: [
  'post', 'put', 'delete', 'get', 'head', 'patch',
  { name: '(other)', value: null }
 ]
```